### PR TITLE
Extracted testing code to separate class.

### PR DESCRIPTION
**Pull request by ljelinkova**
Extracted testing code to separate class - this allows to create various
testing strategies for different release strategies.

### CODE REVIEW

File: ee_tests/src/quickstart_deployments.spec.ts [hunk 1]
```diff
@@ -128,93 +136,13 @@ describe('Creating new quickstart in OSIO', () => {
...
-      expect(cpu.getActualValue()).toBeLessThanOrEqual(cpu.getMaximumValue(), 'the actual cpu usage data has to be lower or equal to maximum');
-
-      let memory = stageDataItems[1];
-      expect(memory.getActualValue()).toBeGreaterThan(0, 'the actual memory usage data has to be greater than 0');
-      expect(memory.getActualValue()).toBeLessThanOrEqual(memory.getMaximumValue(), 'the actual memory usage data has to be lower or equal to maximum');
-    }
+    await deploymentsSpecStrategy.showDeploymentsScreen();
+    let application = await deploymentsSpecStrategy.verifyApplication();
+    await deploymentsSpecStrategy.verifyEnvironments(application);
+    await deploymentsSpecStrategy.verifyResourceUsage();
```
Comment (ppitonak) [inline ee_tests/src/quickstart_deployments.spec.ts]: I don't like the name... we are not verifying resource usage of a strategy but rather of application/deployment/pipeline.

File: ee_tests/src/quickstart_deployments.spec.ts [hunk 2]
```diff
@@ -3,6 +3,9 @@ import { WebDriver, error as SE } from 'selenium-webdriver';
 
 import * as support from './support';
 import { Quickstart } from './support/quickstart';
+import { ReleaseStrategy} from './support/release_strategy';
```
Comment (ppitonak) [inline ee_tests/src/quickstart_deployments.spec.ts]: imported but never used

File: ee_tests/src/spec_strategy/deployment_spec_strategy.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
+// tslint:disable:max-line-length
```
Comment (ppitonak) [inline ee_tests/src/spec_strategy/deployment_spec_strategy.ts]: remove

File: ee_tests/src/spec_strategy/deployment_spec_strategy.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        // nothing to test here
+    }
+}
+
+export class StageDeploymentStrategy extends ReleaseDeploymentStrategy {
+    protected async testEnvironmentsInternal(environments: DeployedApplicationEnvironment[]) {
+        await super.testEnvironmentsInternal(environments);
+        support.info('Verifying application\'s stage environment');
+
+        let environment = environments[Environment.STAGE];
```
Comment (ppitonak) [inline ee_tests/src/spec_strategy/deployment_spec_strategy.ts]: Is the `Environment` the same thing as a `ReleaseStrategy`?

File: ee_tests/src/spec_strategy/deployment_spec_strategy.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+export abstract class DeploymentStrategy {
+
+    protected spaceName: string;
+
+    protected spacePipelinePage: SpacePipelinePage;
+
+    protected spaceDeploymentsPage: SpaceDeploymentsPage;
+
+    constructor(spaceName: string, spacePipelinePage: SpacePipelinePage) {
```
Comment (ppitonak) [inline ee_tests/src/spec_strategy/deployment_spec_strategy.ts]: why do you need `spacePipelinePage` here?

File: ee_tests/src/spec_strategy/deployment_spec_strategy.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        this.spaceDeploymentsPage = new SpaceDeploymentsPage();
+    }
+
+    public async verifyApplication(): Promise<DeployedApplication> {
+        support.info('Verifying deployed applications');
+        let applications = await this.spaceDeploymentsPage.getDeployedApplications();
+        expect(applications.length).toBe(1, 'number of deployed applications');
+
+        let application = applications[0];
+        expect(application.getName()).toBe(this.spaceName, 'application name');
```
Comment (ppitonak) [inline ee_tests/src/spec_strategy/deployment_spec_strategy.ts]: Not necessary in this PR but shouldn't we make this class more general, i.e. no hard-coded values but rather pass these expected values to the function `verifyApplication(1, spaceName)`

File: ee_tests/src/spec_strategy/deployment_spec_strategy.ts (hunk 4 above)
Comment (ljelinkova) [reply ee_tests/src/spec_strategy/deployment_spec_strategy.ts]: Those are 2 independent things.

**Pull request merged**